src/ostorlab/serve_app/oxo.py

```diff
@@ -793,6 +794,60 @@ def mutate(
 
         runtime_instance: runtime.LocalRuntime = runtime.LocalRuntime()
         runtime_instance.follow = []
+        created_scan = runtime_instance.prepare_scan(
+            assets=scan_assets, title=scan.title
+        )
+        RunScanMutation._persist_scan_info(created_scan, scan)
+
+        executor = futures.ThreadPoolExecutor(max_workers=1)
```

Comment:
ThreadPoolExecutor for a single thread, why not use a single thread? or you can pool executor but assign that in the init and use it for all tasks.